plugins/woocommerce/client/blocks/assets/js/blocks/product-elements/add-to-cart-form/frontend.ts

```diff
@@ -1,21 +1,57 @@
 /**
  * External dependencies
  */
-import { store } from '@wordpress/interactivity';
+import { store, getContext } from '@wordpress/interactivity';
 import { HTMLElementEvent } from '@woocommerce/types';
 
+export type Context = {
+	productId: number;
+	productType: string;
+	quantity: Record< number, number >;
+	childProductId: number;
+	quantityConstraints: Record<
+		number,
+		{ min: number; max: number | null; step: number }
+	>;
+};
+
 const getInputElementFromEvent = (
 	event: HTMLElementEvent< HTMLButtonElement >
 ) => {
 	const target = event.target as HTMLButtonElement;
 
 	const inputElement = target.parentElement?.querySelector(
-		'.input-text.qty.text'
-	) as HTMLInputElement | null | undefined;
+		'.wc-block-components-quantity-selector__input'
+	) as HTMLInputElement | null;
 
 	return inputElement;
 };
 
+const getQuantityStateInfo = () => {
+	const context = getContext< Context >();
+	const {
+		quantity,
+		childProductId,
+		productType,
+		quantityConstraints,
+		productId,
+	} = context;
+	const id =
+		productType === 'grouped' && childProductId
+			? childProductId
+			: productId;
+	const constraints = quantityConstraints?.[ id ] || {
+		min: productType === 'grouped' && childProductId ? 0 : 1,
+		step: 1,
+		max: null,
+	};
+	const currentQuantity =
+		productType === 'grouped' && childProductId
+			? quantity?.[ childProductId ] || 0
+			: quantity?.[ productId ] || 0;
```

Comment:
Same here, this can be:

```suggestion
	const currentQuantity = quantity?.[ id ] || 0;
```